### DISCUSSION TITLE
Moving from ZwaveJS2MQTT to zwave-js-ui SNAP package channel

### DISCUSSION THREAD

**Discussion by chench00** [zwave-js/zwave-js-ui#2670] [Q&A]
Pardon the question in advance but what is the proper way to handle the rename if I am using the ZwaveJS2MQTT snap package?

**Comment by Thaui57** [comment]
Interesting question. I also need some feedback how to move to the new package. Some tests gave a big mess in my database

**Comment by robertsLando** [comment]
@jmgiaever Is maintaining it so I let him answer to this question :)

**Comment by jmgiaever** [comment]
Hi,

It will be released under a new name, `zwave-js-ui`. There will be a connection between the two snaps, so you easily can move/copy from z2m to zui. 

Unfortunately I haven't had time to finish the code changes within the snap from z2m to zui yet, but will do it this week.

**Comment by Thaui57** [comment]
Thanks, will try it after your changes. Existing users need some advice how to change to zwave-js-ui not losing their data. BRAm 26.09.2022 09:24 schrieb Joachim Marthinsen Giæver ***@***.***>:
Hi,
It will be released under a new name, zwave-js-ui. There will be a connection between the two snaps, so you easily can copy from z2m to zui.
Unfortunately I haven't had time to finish the code changes within the snap from z2m to zui yet, but will do it this week.

**Comment by jmgiaever** [comment]
Yeah, I haven't gotten that far to completely implement it, but (if everything goes as I want to) then zui will become a requirement of z2m and pull the configurations from z2m upon installation, so the user won't have to do anything technical to port to it.

**Comment by jordahlm** [comment]
Tossing my support on this. Eagerly awaiting any update even if it's just some pointers on what data to manually copy into the folder structure.

**Comment by jmgiaever** [comment]
https://github.com/giaever-online-iot/zwavejs2mqtt
https://github.com/giaever-online-iot/zwave-js-ui

You can watch these repositories. Both are building at the moment; however one of them to offer support for transitioning to the other. 

And sorry for it to have taken so much time, but it's just me maintaining this package, and it couldn't come at an more unpleasant time regards to work.

This is my current position on the map (traveling with work): ![Screenshot_20221005-215348.png](https://user-images.githubusercontent.com/7188121/194150665-6da5adf0-4171-42a7-af3a-e271c9cc0bfd.png)

I still have some testing and verifying to do.

**Reply by chench00** [reply]
Looking forward to it as I just discovered that the latest version of HA requires me to upgrade to zwave-js-ui.

**Comment by jordahlm** [comment]
For what it's worth, it wasn't too painful to manually move to the new snap. I exported my nodes and settings and also manually copied out my important settings to a text file. Then I deactivated the old snap and activated the new one. For some reason it wouldn't import the settings from the file, but was easy enough to to configure manually. I was able to import my nodes with no issue. After that I just had to wait for all my nodes to be re-interviewed. As the actual pairing is maintained in the NVRAM of the z-wave stick itself there is not too much risk of having to re-pair your entire network. On my network of around 40 nodes it took about 30 minutes to finish interviewing all of the hardwired nodes, and then I had to manually wake up a few battery powered nodes to get them to complete the new interview.

**Reply by chench00** [reply]
I thought about moving it myself but afraid of screwing things up. Any way you can put some instructions together for what worked for you?

**Reply by jmgiaever** [reply]
https://github.com/zwave-js/zwave-js-ui/issues/2729#issuecomment-1277476146

**Reply by jmgiaever** [reply]
Hi,

Z2M should automatically install ZUI, convert configs to ZUI and enable it. 

BUT only if you haven't connected serial-port interface. Then it won't enable. Just install and convert it. 

This is because serial-port need some specifics on which serial-port.
